src/libs/portal.ts

```diff
@@ -336,48 +378,26 @@ export async function getFromCalCalendar(db: Db, user: string) {
 /**
  * Retrieves the day rotation for a given date.
  * @param date The date to get the rotation for, defaults to today.
- * @returns The day rotation (integer in [1, 6]).
+ * @returns The day rotation (character A-H).
  */
 export async function getDayRotation(date: Date) {
-	const scheduleDate = new Date(date);
-
-	let body;
-	try {
-		body = await request(dayRotationURL);
-	} catch (e) {
-		throw new Error('There was a problem fetching the day rotation!');
-	}
-
-	const data = ical.parseICS(body);
-
-	// School Portal does not give a 404 if calendar is invalid. Instead, it gives an empty calendar.
-	// Unlike Canvas, the portal is guaranteed to contain some sort of data within a span of a year.
-	if (_.isEmpty(data)) {
-		throw new Error('There was a problem fetching the day rotation!');
-	}
-
-	for (const calEvent of Object.values(data)) {
-		if (typeof calEvent.summary !== 'string') {
-			continue;
-		}
-
-		const start = new Date(calEvent.start!);
-		const end = new Date(calEvent.end || '');
-
-		const startTime = start.getTime();
-		const endTime = end.getTime();
-
-		// Check if it's an all-day event
-		if (startTime === scheduleDate.getTime() && Number.isNaN(endTime)) {
-			// See if valid day
-			if (validDayRotationPlain.test(calEvent.summary)) {
-				// Get actual day
-				return parseInt(/Day ([1-6])/.exec(calEvent.summary)![1], 10);
-			}
-		}
-	}
+	return withCalSummary(date, (summary: string) => {
```

Comment:
Can you remove the type annotation? Also look at making this just a single-line arrow function, but feel free to keep it as bracketed if it's too ugly